common/lib/xmodule/xmodule/js/src/capa/display.coffee

```diff
@@ -150,15 +150,24 @@ class @Problem
     $.postWithPrefix "#{url}/input_ajax", data, callback
 
 
-  render: (content) ->
-    if content
+  # Like render(), but for use when we already have our content up front and
+  # don't need to load it from across the network. If content is supplied (e.g.
+  # it was in a data- attribute or the caller wants to replace the content),
+  # then we use that. Otherwise, we assume that the proper content has already
+  # been loaded into @el by the server side rendering.
+  preRender: (content) ->
       @el.attr({'aria-busy': 'true', 'aria-live': 'off', 'aria-atomic': 'false'})
-      @el.html(content)
+      if content
+        @el.html(content)
       JavascriptLoader.executeModuleScripts @el, () =>
         @setupInputTypes()
         @bind()
         @queueing()
       @el.attr('aria-busy', 'false')
+
+  render: (content) ->
+    if content
+      @preRender(content)
```

Comment:
This a little confusing to me.

It looks like `preRender` is duplicating what `render` does, partially.  It seems like `render` ought to have the same signature, but `preRender` should be the check where if there's no content passed in, it does the AJAX call which then goes to `render`.

Maybe I'm not understanding what `@forceUpdate` does, though.

Does my comment make sense?  A code example may explain it better.